pkg/plugins/golang/v3/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -83,9 +83,11 @@ type {{ .Resource.Kind }}Reconciler struct {
 	Scheme *runtime.Scheme
 }
 
+{{ if not (isEmptyStr .Resource.Path) -}}
 //+kubebuilder:rbac:groups={{ .Resource.QualifiedGroup }},resources={{ .Resource.Plural }},verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups={{ .Resource.QualifiedGroup }},resources={{ .Resource.Plural }}/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups={{ .Resource.QualifiedGroup }},resources={{ .Resource.Plural }}/finalizers,verbs=update
+{{- end }}
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
```

Comment:
Hi @kunalp-gohire, 

Thank you for the explanation and collaboration. 🥇 


The testdata/ dir samples are generated based on the source code (pr changes). In this case, see that we already both scenarios for we verify your change:

- (Reconcile an external type Laker) https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/testdata/generate.sh#L72

- (Reconcile a core type Deployments) https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/testdata/generate.sh#L96

_Note that you must run `make generate` to re-generate the samples._ 

**Then, why we would not require the permissions to have reconcile the external type (Laker)?**
*external type: It is for e.g an API/CRD/Kind defined in Project A but used the Project B

Following are some comments inline:

> As markers are required to generate roles/permissions to operate a particular resource or kind. If we are not creating any resource and just created a controller to reconcile or watch another resource so we manually add markers for that resource so that after making manifests config/rbac should update with correct permissions. As in this scenario, we haven't created a resource for the controller but still, markers for that resource have been created and

Are you:
a) Creating an API/Kind and a controller for this kind
b) Then, create another controller for the same kind defined above

Is that the scenario? 

>  after applying make manifests the permissions ( redundant permissions) are generated.

Could not it be related to : https://github.com/kubernetes-sigs/controller-tools/pull/620

Reply:
@camilamacedo86  There could be 2 scenarios:
1] Create a ‘xyz’ controller to reconcile or watch the ‘abc’ resource which is created in the same project.  

2] Create a ‘xyz’ controller to reconcile or watch the ‘abc’ resource (custom/ standard) 
which is created in another project.

In both cases the markers will be generated for resource ‘xyz’ but there is no resource named ‘xyz’ defined in the project. 


In test data you are creating a controller / api named Laker  without resource  `(create api --group crew --version v1 --kind Laker --controller=true --resource=false --make=false   ) `to reconcile resource Laker which is defined in another project.  In this case markers will be generated for resource Laker as here controller name is same as external type resource Laker.
 But one can create an api/controller ‘sample’ to reconcile external type resources ‘Laker’, ‘Ship’, ect.   
`(create api --group crew --version v1 --kind Sample --controller=true --resource=false --make=false   ) `In this case markers for a sample resource will be created which is not created or defined in the project. And makers for resources Laker, Ship and other resources which we intend to use in a sample controller need to be added manually.


In my opinion markers for api/controller for which resource is not created shouldn’t be generated. `(--resource=false --make=false   )` As there might be a possibility that the controller name and external resource name for which we are creating the controller might not be the same.